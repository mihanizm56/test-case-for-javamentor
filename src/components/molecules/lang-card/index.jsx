import React from 'react'
import './lang-card.css'

export const LangCard = ({
    title, createdAt, numberOfProjects, link, logo
}) => {
    console.log('check LangCard', title, createdAt, numberOfProjects, link, logo)

    return (
        <div className='card-container'>
            <div className='card-first-line'>
                <img alt='logo' src={logo} height='118px' width='118px' />
            </div>
            <div className='card-second-line'>
                <span className='card__lang-name'>{title}</span>
                <span className='card__created'>
                    {`Основан в ${createdAt} году`}
                </span>
                <span className='card__number-of-projects'>
                    {`Проектов на GitHub ${numberOfProjects} году`}
                </span>
                <a className='card__link' href={link}>
                  Документация
                </a>
            </div>
        </div>
    )
}

LangCard.defaultProps = {}
