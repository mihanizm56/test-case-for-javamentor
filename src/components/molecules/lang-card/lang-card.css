.card-container {
    margin-top: 10px;
    height: 120px;
    display: flex;
    padding: 10px;
    border: 1px solid black;
}

.card-first-line {
    height: 118px;
    width: 118px;
}

.card-second-line {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 0 0 20px 10px;
    width: 230px;
}

.card__lang-name {
    font-family: 'Verdana-Bold', "Arial", 'sans-serif';
    font-size: 15px;
}

.card__created, .card__number-of-projects, .card__link {
    font-family: 'Verdana', "Arial", 'sans-serif';
    font-size: 15px;
    word-wrap: break-word;
}