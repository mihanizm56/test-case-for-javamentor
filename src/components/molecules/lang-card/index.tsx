import React, {memo} from 'react'
import './lang-card.css'

interface LangCardType {
    title:string, 
    createdAt:number, 
    numberOfProjects:number, 
    link:string, 
    logo:string
}

export const LangCard = memo<LangCardType>(({
    title, createdAt, numberOfProjects, link, logo
}) => (
        <div className='card-container'>
            <div className='card-first-line'>
                <img alt='logo' src={logo} height='118px' width='118px' />
            </div>
            <div className='card-second-line'>
                <span className='card__lang-name'>{title}</span>
                <span className='card__created'>
                    {`Основан в ${createdAt} году`}
                </span>
                <span className='card__number-of-projects'>
                    {`Проектов на GitHub ${numberOfProjects} году`}
                </span>
                <a className='card__link' href={link}>
                  Документация
                </a>
            </div>
        </div>
    )
)
