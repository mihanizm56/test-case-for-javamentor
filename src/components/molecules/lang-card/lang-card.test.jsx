import React from "react";
import { shallow } from "enzyme";
import { LangCard } from "../lang-card";

describe("test LangCard", () => {
  const cardProps = {
    title: "test_title",
    createdAt: 1,
    numberOfProjects: 1,
    link: "test_link",
    logo: "test_logo"
  };
  let component;

  beforeEach(() => {
    jest.clearAllMocks();
    component = shallow(<LangCard {...cardProps} />);
  });

  describe("test render contract", () => {
    it("renders html-content", () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe("test props contract", () => {
    it("check title prop", () => {
      expect(component.find(".card__lang-name").text()).toEqual("test_title");
    });
    it("check createdAt prop", () => {
      expect(component.find(".card__created").text()).toEqual(
        `Основан в ${cardProps.createdAt} году`
      );
    });
    it("check numberOfProjects prop", () => {
      expect(component.find(".card__number-of-projects").text()).toEqual(
        `Проектов на GitHub ${cardProps.numberOfProjects}`
      );
    });
    it("check link prop", () => {
      expect(component.find(".card__link").prop("href")).toEqual(
        cardProps.link
      );
    });
    it("check logo prop", () => {
      expect(component.find("img").prop("src")).toEqual(cardProps.logo);
    });
  });
});
