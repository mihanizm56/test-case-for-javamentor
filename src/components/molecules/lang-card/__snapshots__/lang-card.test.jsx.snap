// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LangCard test render contract renders html-content 1`] = `
<div
  className="card-container"
>
  <div
    className="card-first-line"
  >
    <img
      alt="logo"
      height="118px"
      src="test_logo"
      width="118px"
    />
  </div>
  <div
    className="card-second-line"
  >
    <span
      className="card__lang-name"
    >
      test_title
    </span>
    <span
      className="card__created"
    >
      Основан в 1 году
    </span>
    <span
      className="card__number-of-projects"
    >
      Проектов на GitHub 1
    </span>
    <a
      className="card__link"
      href="test_link"
    >
      Документация
    </a>
  </div>
</div>
`;
